### PR TITLE
Adds update orcid work endpoint support.

### LINKED ISSUE #360: Support create/update orcid work endpoint
See https://github.com/sul-dlss/dor-services-app/issues/4533

### PR DESCRIPTION
**Pull request by justinlittman**
closes #360

## Why was this change made? 🤔
Support endpoint for use by robots.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: lib/dor/services/client/object.rb [hunk 1]
```diff
@@ -106,6 +106,18 @@ def update_doi_metadata
           raise_exception_based_on_response!(resp)
         end
 
+        # Update the ORCID Works
+        # @raise [NotFoundResponse] when the response is a 404 (object not found)
+        # @return [boolean] true on success
```
Comment (lwrubel) [inline lib/dor/services/client/object.rb:111]:
```suggestion
        # @return [Boolean] true on success
```

File: lib/dor/services/client/object.rb [hunk 2]
```diff
@@ -106,6 +106,18 @@ def update_doi_metadata
           raise_exception_based_on_response!(resp)
         end
 
+        # Update the ORCID Works
```
Comment (lwrubel) [inline lib/dor/services/client/object.rb:109]: Will this method be updating multiple ORCID works?

**Comment by lwrubel** [comment]
Approved pending consideration of YARD doc question.

File: lib/dor/services/client/object.rb (hunk 2 above)
Comment (justinlittman) [reply lib/dor/services/client/object.rb:109]: So, that's really a metaphysical question as to what constitutes a "work" in the orcid system. However, I have been using the singular elsewhere, so fixing here.

**Pull request merged**